src/actions/operator.ts

```diff
@@ -1094,21 +1096,22 @@ class ActionVisualReflowParagraph extends BaseOperator {
   }
 
   public async run(vimState: VimState, start: Position, end: Position): Promise<VimState> {
-    start = Position.EarlierOf(start, end);
-    end = Position.LaterOf(start, end);
+    const front = Position.EarlierOf(start, end);
+    const back = Position.LaterOf(start, end);
 
-    let textToReflow = TextEditor.getText(new vscode.Range(start, end));
-    let indent = this.getIndentation(textToReflow);
+    start = front.getLineBegin();
+    end = back.getLineEnd();
 
-    textToReflow = this.reflowParagraph(textToReflow, indent);
+    let textToReflow = TextEditor.getText(new vscode.Range(start, end));
+    textToReflow = this.reflowParagraph(textToReflow);
 
     vimState.recordedState.transformations.push({
       type: 'replaceText',
       text: textToReflow,
       start: start,
       end: end,
-      // Move cursor to front of line to realign the view
-      diff: PositionDiff.newBOLDiff(),
+      // Move cursor to front of lowest line to realign the view
+      diff: PositionDiff.newBOLDiff(textToReflow.split('\n').length - 1),
```

Comment:
Is it safe to just split by `\n`? I mean can this handle `\r\n`.
Hope I am just over thinking.

Reply:
Yes. `"\n".split('\n')` and `"\r\n".split('\n')` will return the same result. You can check that right now by pressing f12. :-P